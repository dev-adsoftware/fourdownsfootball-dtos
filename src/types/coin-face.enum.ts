/* eslint-disable no-shadow */
export enum CoinFace {
  Heads,
  Tails,
}
