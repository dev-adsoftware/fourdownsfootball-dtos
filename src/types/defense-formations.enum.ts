export enum DefenseFormations {
  KickReturn = 'kick-return',
}
