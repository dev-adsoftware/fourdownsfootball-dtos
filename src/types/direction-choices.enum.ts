export enum DirectionChoices {
  Home,
  Away,
}
