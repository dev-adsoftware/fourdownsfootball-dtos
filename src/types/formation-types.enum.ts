export enum FormationTypes {
  Kickoff = 'kickoff',
  KickReturn = 'kick-return',
  Offense = 'offense',
  Defense = 'defense',
  Punt = 'punt',
  PuntReturn = 'punt-return',
  FieldGoal = 'field-goal',
  FieldGoalDefense = 'field-goal-defense',
}
