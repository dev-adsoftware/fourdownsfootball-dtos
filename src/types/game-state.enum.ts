export enum GameState {
  AwaitingCoinFaceChoice,
  AwaitingCoinFlipSim,
  AwaitingKickoffChoice,
  AwaitingDirectionChoice,
  AwaitingKickPlaycall,
  AwaitingReceivingPlaycall,
  AwaitingOffensePlaycall,
  AwaitingDefensePlaycall,
  AwaitingPlaySim,
}
