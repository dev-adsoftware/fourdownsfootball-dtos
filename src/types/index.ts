export * from './coin-face.enum';
export * from './direction-choices.enum';
export * from './formation-types.enum';
export * from './game-state.enum';
export * from './kickoff-choices.enum';
export * from './offense-formations.enum';
export * from './defense-formations.enum';
export * from './random-player-types.enum';
