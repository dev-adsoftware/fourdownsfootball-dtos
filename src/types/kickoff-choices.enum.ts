export enum KickoffChoices {
  Defer,
  Kickoff,
  Receive,
}
