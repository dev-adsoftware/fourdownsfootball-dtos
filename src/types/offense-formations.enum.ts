export enum OffenseFormations {
  Kickoff = 'kickoff',
}
