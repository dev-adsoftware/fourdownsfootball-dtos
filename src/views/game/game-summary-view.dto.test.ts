import { GameSummaryView } from '.';
import { DtoFactory } from '../..';
import { GameState } from '../../types';

describe('given: game summary data', () => {
  describe('when: I create a game summary view with a dto factory', () => {
    it('then: game summary view was returned', async () => {
      const view = DtoFactory.create(GameSummaryView, {
        id: 'jest.id',
        sequence: '0',
        state: GameState.AwaitingCoinFaceChoice,
        actor: 'jest.actor',
        homeUsername: 'jest.homeUsername',
        homeTeamId: 'jest.homeTeamId',
        awayUsername: 'jest.awayUsername',
        awayTeamId: 'jest.awayTeamId',
      });
      expect(view).toEqual({
        actor: 'jest.actor',
        awayScore: 0,
        awayTeamId: 'jest.awayTeamId',
        awayUsername: 'jest.awayUsername',
        ballOn: 25,
        down: 1,
        gameClock: 0,
        homeScore: 0,
        homeTeamId: 'jest.homeTeamId',
        homeUsername: 'jest.homeUsername',
        id: 'jest.id',
        sequence: '0',
        state: 0,
        toGo: 10,
      });
    });
  });
});
