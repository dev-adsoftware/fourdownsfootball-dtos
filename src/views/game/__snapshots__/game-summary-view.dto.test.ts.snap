// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given: missing data when: I initialize then: error matched snapshot 1`] = `[Error: GameSummaryView: id must be a string, sequence must be a number string, currentSeed must be a number conforming to the specified constraints, homeTeam must be an object, awayTeam must be an object, stateAttributes must be an object]`;

exports[`given: missing nested data when: I initialize then: error matched snapshot 1`] = `
[Error: GameSummaryView: id must be a string, sequence must be a number string, currentSeed must be a number conforming to the specified constraints, An instance of an object has failed the validation:
 - property homeTeam.id has failed the following constraints: isString 
 - property homeTeam.sequence has failed the following constraints: isNumberString 
 - property homeTeam.ownerId has failed the following constraints: isString 
 - property homeTeam.attributes has failed the following constraints: isObject 
, An instance of an object has failed the validation:
 - property awayTeam.id has failed the following constraints: isString 
 - property awayTeam.sequence has failed the following constraints: isNumberString 
 - property awayTeam.ownerId has failed the following constraints: isString 
 - property awayTeam.attributes has failed the following constraints: isObject 
, An instance of an object has failed the validation:
 - property stateAttributes.state has failed the following constraints: isEnum 
 - property stateAttributes.actorId has failed the following constraints: isString 
]
`;
